BuildResidentialHPXML/measure.rb

```diff
@@ -3551,14 +3568,40 @@ def self.create_geometry_envelope(runner, model, args)
     return true
   end
 
-  def self.set_header(hpxml, args)
+  def self.unavailable_period_exists(hpxml, column_name, begin_month, begin_day, begin_hour, end_month, end_day, end_hour, natvent_availability = nil)
+    hpxml.header.unavailable_periods.each do |unavailable_period|
+      begin_hour = 0 if begin_hour.nil?
+      end_hour = 24 if end_hour.nil?
+
+      next unless (unavailable_period.column_name == column_name) &&
+                  (unavailable_period.begin_month == begin_month) &&
+                  (unavailable_period.begin_day == begin_day) &&
+                  (unavailable_period.begin_hour == begin_hour) &&
+                  (unavailable_period.end_month == end_month) &&
+                  (unavailable_period.end_day == end_day) &&
+                  (unavailable_period.end_hour == end_hour)
+      if unavailable_period.natvent_availability == natvent_availability
+        return true
+      end
+    end
+    return false
+  end
+
+  def self.set_header(runner, hpxml, args)
+    errors = []
+
     hpxml.header.xml_type = 'HPXML'
     hpxml.header.xml_generated_by = 'BuildResidentialHPXML'
     hpxml.header.transaction = 'create'
 
     if args[:schedules_vacancy_period].is_initialized
       begin_month, begin_day, begin_hour, end_month, end_day, end_hour = Schedule.parse_date_time_range(args[:schedules_vacancy_period].get)
-      hpxml.header.unavailable_periods.add(column_name: 'Vacancy', begin_month: begin_month, begin_day: begin_day, begin_hour: begin_hour, end_month: end_month, end_day: end_day, end_hour: end_hour, natvent_availability: HPXML::ScheduleUnavailable)
+
+      natvent_availability = HPXML::ScheduleUnavailable
```

Comment:
Seems like this should be inside the `unavailable_period_exists` method and not here.

Reply:
I'm not sure I understand. It gets passed into the `unavailable_period_exists` method. For the "Vacancy" instance, it's always set to `natvent_availability = HPXML::ScheduleUnavailable`. For "Power Outage", it's assigned using an argument.

Reply:
So inside the method, set `natvent_availability = HPXML::ScheduleUnavailable if natvent_availability.nil?`?

Reply:
Yeah, I guess. We'd then be assigning `natvent_availability = HPXML::ScheduleUnavailable` twice for the "Vacancy" instance...